### PR TITLE
feat(helix-view): dynamic line numbers

### PR DESCRIPTION
**Pull request by The0x539**
This is a configuration that I've carried with me from Vim to Neovim to Kakoune.

This implementation was remarkably easy to figure out; in the past, it had always been unusually difficult and relied on various mode change hooks and what not.

### CODE REVIEW

**Comment by pickfire** [comment]
Can I know how did you do this in vim, neovim, and kakoune?

Although this seemed useful as a default (at least from what I see given that relative number is only showing at the time it is useful), it seemed oddly specific.

**Review by pickfire — APPROVED** [review]

**Comment by pickfire** [comment]
I think you also need to update the book `configuration.md` but if you just put `dynamic` I doubt readers understand what it meant.

**Comment by The0x539** [comment]
> Can I know how did you do this in vim, neovim, and kakoune?

I'm pretty sure I originally found my original Vim version online on some "tips and tricks" wiki that I could swear was on Wikia, but I can't find it now. Maybe I'm misremembering and never had it in regular Vim; a quick search of the `master` (old) branch of my dotfiles repo suggests as much.

[This](https://github.com/The0x539/dotfiles/blob/757153367bcf0b3f3c3e73796ec90c5e70c69614/.config/nvim/init.vim#L36-L41), after some refinement, is how I did it in Neovim, and [this mess](https://github.com/The0x539/dotfiles/blob/5b595661e87967c10ac286323120c7543516e8ef/.config/kak/ruler.kak), of my own invention, is how I do it in Kakoune.

> I think you also need to update the book `configuration.md`

Yes, good catch. I'll do that now.

File: book/src/configuration.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ To override global configuration parameters, create a `config.toml` file located
 | `middle-click-paste` | Middle click paste support. | `true` |
 | `scroll-lines` | Number of lines to scroll per scroll wheel step. | `3` |
 | `shell` | Shell to use when running external commands. | Unix: `["sh", "-c"]`<br/>Windows: `["cmd", "/C"]` |
-| `line-number` | Line number display (`absolute`, `relative`) | `absolute` |
+| `line-number` | Line number display: `absolute` simply shows each line's number, `relative` shows the distance from the current line, and `dynamic` switches between the two based on whether the editor is unfocused or in insert mode. | `absolute` |
```
Comment (pickfire) [inline book/src/configuration.md:19]: Should we call it hybrid like the original article rather than dynamic?

Comment (The0x539) [reply book/src/configuration.md:19]: Hybrid refers to the behavior added (unconditionally, not that it bothers me) in #656, not this contribution.

**Comment by archseer** [comment]
Should we just make this the default behavior of relative line number? I can't imagine you'd want relative line numbers in unfocused windows

**Comment by archseer** [comment]
Let's make this the default behavior of `relative`!

**Pull request merged**